app/src/org/commcare/views/media/MediaLayout.java

```diff
@@ -114,69 +110,75 @@ public static MediaLayout buildComprehensiveLayout(Context context,
         return mediaLayout;
     }
 
+    public void addDivider() {
+        divider.setVisibility(VISIBLE);
+    }
+
+    //region private helpers
+
     private void setAVT(TextView text, String audioURI, String imageURI,
                         final String videoURI, final String bigImageURI,
                         final String qrCodeContent, String inlineVideoURI,
                         boolean showImageAboveText,
                         int questionIndex) {
-        viewText = text;
-        mInlineVideoUri = inlineVideoURI;
-        mImageURI = imageURI;
-        mBigImageURI = bigImageURI;
-        mQrCodeContent = qrCodeContent;
-
-        RelativeLayout questionTextPane = new RelativeLayout(this.getContext());
-        questionTextPane.setId(QUESTION_TEXT_PANE_ID);
-
         setupStandardAudio(audioURI, questionIndex);
         setupVideoButton(videoURI);
 
-        // Now set up the center view -- it is either an image, a QR Code, an inline video, or
-        // expanded audio
-        mediaPane = new RelativeLayout(getContext());
+        // We only show one of the inline-video-view / qrview / image
+        if (inlineVideoURI != null) {
+            setupInlineVideoView(inlineVideoURI);
+        } else if (qrCodeContent != null) {
+            setupQRView(qrCodeContent);
+        } else if (imageURI != null) {
+            setupImage(imageURI, bigImageURI);
+        }
+
+        addTextView(text);
+        if (showImageAboveText || DeveloperPreferences.imageAboveTextEnabled()) {
+            showMediaAboveText();
+        }
+    }
+
+    private void showMediaAboveText() {
+        // This step will change the layout constraints of the views in xml to shift
+        // media(video, image qr ) above text.
 
-        LayoutParams mediaPaneParams = refreshMediaView();
+        // Change the questionBottomBarrier to align at the top of parent which would automatically
+        // align the mediaViews(including missing media view) to the top.
+        questionBottomBarrier.setReferencedIds(new int[]{ R.id.top_guideline });
 
-        addAudioVideoButtonsToView(questionTextPane);
+        // Next align the text, audiobutton below mediaView.
+        alignTextContainerBelowMediaView(audioButton);
```

Comment:
Wondering if it's possible to avoid multiple calls of `alignTextContainerBelowMediaView` as well and instead use barrier/guideline mechanism to shift these views ?